src/sysupdate/updatectl.c

```diff
@@ -0,0 +1,1218 @@
+}
+
+static int verb_update(int argc, char **argv, void *userdata) {
+        sd_bus *bus = ASSERT_PTR(userdata);
+        _cleanup_(sd_event_unrefp) sd_event *event = NULL;
+        _cleanup_(sd_event_source_unrefp) sd_event_source *render_exit = NULL;
+        _cleanup_ordered_hashmap_free_ OrderedHashmap *map = NULL;
+        _cleanup_strv_free_ char **targets = NULL, **versions = NULL;
+        _cleanup_strv_free_ char **target_paths = NULL;
+        size_t n;
+        unsigned remaining = 0;
+        void *p;
+        bool did_anything = false;
+        int r;
+
+        r = ensure_targets(bus, argv + 1, &targets);
+        if (r < 0)
+                return r;
+
+        r = parse_targets(targets, &n, &target_paths, &versions);
+        if (r < 0)
+                return r;
+
+        map = ordered_hashmap_new(&string_hash_ops_free);
+        if (!map)
+                return log_oom();
+
+        r = sd_event_default(&event);
+        if (r < 0)
+                return log_error_errno(r, "Failed to get event loop: %m");
+
+        r = sd_bus_attach_event(bus, event, 0);
+        if (r < 0)
+                return log_error_errno(r, "Failed to attach bus to event loop: %m");
+
+        r = sd_event_set_signal_exit(event, true);
+        if (r < 0)
+                return log_error_errno(r, "Failed to set up interrupt handler: %m");
+
+        for (size_t i = 0; i < n; i++) {
+                _cleanup_(userdata_freep) Userdata *u = NULL;
+                u = userdata_new(map, bus, &remaining, target_paths[i], targets[i]);
+                if (!u)
+                        return log_oom();
+
+                r = sd_bus_call_method_async(bus, NULL, bus_sysupdate_mgr->destination, target_paths[i], SYSUPDATE_TARGET_INTERFACE, "Update", update_started, u,
+                                          "s", versions[i]);
+                if (r < 0)
+                        return r;
+                TAKE_PTR(u);
+
+                remaining++;
+        }
+
+        /* Set up the rendering */
+        r = sd_event_add_post(event, NULL, update_render_progress, map);
+        if (r < 0)
+                return r;
+
+        r = sd_event_add_exit(event, &render_exit, update_render_progress, map);
+        if (r < 0)
+                return r;
+
+        r = sd_event_source_set_priority(render_exit, SD_EVENT_PRIORITY_IDLE);
+        if (r < 0)
+                return r;
+
+        r = sd_event_loop(event);
+        if (r < 0)
+                return r;
+
+        ORDERED_HASHMAP_FOREACH(p, map) {
+                r = PTR_TO_INT(p);
+                if (r == -EALREADY)
+                        continue;
+                if (r == UPDATE_PROGRESS_FAILED)
+                        return EXIT_FAILURE;
+                if (r < 0)
+                        return r;
+
+                did_anything = true;
+        }
+
+        if (arg_reboot) {
+                if (did_anything)
+                        return reboot_now();
+                else
```

Comment:
this `else` can be removed without any effect, hence please do.

saves you indentation on the next line, and we like shallow indentation